### DISCUSSION TITLE
Dropbox 无法授权的解决方案，包括：闪退与提示密码过长

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#3171] [Developer]
## 最新说明

经过确认 Dropbox Token 方案彻底变为**临时 Token**，暂时没有任何办法。

如果通过此方式得到的 Token 为 `sl.xxx` 则为临时 Token，但 **已申请过的 Token 仍长期有效**，不受此次改变影响。

## 问题现象

1. 授权 Dropbox 时出现**页面闪退**。

2. 使用 [Dropbox 授权登录](https://github.com/Kenshin/simpread/discussions/1570) 时提示 **密码过长导致无法加密，请使用坚果云授权方案**

   ![image](https://user-images.githubusercontent.com/81074/145162809-dabf8d3b-6a42-4d16-98e8-c669ad1ec2c6.png)

## 问题说明

Dropbox 更改了 API 导致

## 解决方案

> 操作前请注意

1. 在简悦中关闭 Dropbox 授权。

   ![image](https://user-images.githubusercontent.com/81074/145162892-3497bb8f-a4b8-4330-a9d2-9ee6e9de9a3c.png)

2. 在 Dropbox 取消授权，[点击这里](https://www.dropbox.com/account/connected_apps) 取消授权。

> 操作步骤

1. 直接在浏览器打开 [Dropbox 授权链接](https://www.dropbox.com/oauth2/authorize?response_type=token&redirect_uri=https://www.dropbox.com/1/oauth2/display_token&client_id=4cyaw4wqpbg4751)

![image](https://user-images.githubusercontent.com/81074/145160913-1b190ee1-80a7-4428-bd0a-103ae648fa79.png)

2. 确保开启下面的选项，选项页 → 服务

![image](https://user-images.githubusercontent.com/81074/145146015-0a65e64a-3436-4e09-8ab3-1db69d55cbab.png)

3. 如果使用了**同步助手方案**，请在本地找到 `simpread_config.json`；如果没有使用同步助手，请通过选项页 → 共通 → 导出配置文件

![image](https://user-images.githubusercontent.com/81074/145162242-6acffa0b-74c1-4d62-8aec-f92b6590c1a5.png)

4. 用文本编辑器打开它，然后复制内容并张贴到 https://tool.lu/json 再粘贴回  `simpread_config.json`，然后找到 `secret` 替换为你复制的 token

![image](https://user-images.githubusercontent.com/81074/145162321-44135975-6a93-4e38-b3a5-d0840bfeb5ae.png)

5. 导入

    - 使用同步助手的用户，直接刷新选项页即可。
    
    - 没有使用同步助手的用户，请将配置文件导入到简悦即可。（选项页 → 共通 → 从本地导入配置文件）

    ![image](https://user-images.githubusercontent.com/81074/145161300-1a7b29cd-4c10-4bb4-832b-5fbc2cea0ca7.png)

6. 导出成功后，会发现 Dropbox 已经成功授权了。

![image](https://user-images.githubusercontent.com/81074/145162536-b6217f65-41aa-4c12-8634-bab1fec020ae.png)

## 备注

> 如果发现你的 Dropbox 有下面的结构，请删除红框内的内容。

![image](https://user-images.githubusercontent.com/81074/145162006-ae2753e6-2d23-4c69-a75b-c2f5b720c397.png)

## 感谢

上述操作来自 [简悦社区](https://t.me/simpread) 用户 @haokaiyang 的教程。👏

## 关联问题

- https://github.com/Kenshin/simpread/issues/3170

- https://github.com/Kenshin/simpread/issues/3122

- https://github.com/Kenshin/simpread/issues/2996

- https://github.com/Kenshin/simpread/issues/3032

**Comment by hechen** [comment]
👍

**Comment by eallion** [comment]
works

**Comment by JQKingCN** [comment]
用了这个方法使用dropbox授权登录时还是有“密码过长导致无法加密，请使用坚果云授权方案”的错误

**Reply by Kenshin** [reply]
并不会的，请确认你的操作。
如果还是存在问题，请通过 https://github.com/kenshin/simpread/issues/new 问题，而不是这里。

**Reply by Alephant6** [reply]
我也是，按照这个方法做了两次，还是有这个错误

**Reply by Kenshin** [reply]
@Alephant6 @JQKingCN 

嗨，经过确认 Dropbox Token 方案彻底变为临时 Token，暂时没有任何办法，除非付费。

如果通过此方式得到的 Token 为 `sl.xxx` 则为临时 Token，但之前申请的 Token 仍长期有效。

**Reply by yuanweize** [reply]
我也是.

可以支持onedrive,gdrive之类的同步方式吗?在国外不是很想用坚果云,但是只能用坚果云授权登录..... Dropbox可以同步配置,但是不能授权登录,想用tg保存url到md的功能

**Reply by Kenshin** [reply]
@yuanweize 

会接入一些坚果云以外的网盘。

**Reply by Kenshin** [reply]
@yuanweize 

另外，可能需要你帮个忙：我在考虑切换某些 CDN 环境，方便的话，帮忙测试下。

请问你访问 https://cdn.staticfile.org/font-awesome/6.2.0/css/fontawesome.min.css 是否正常？同时请告知速度如何？

**Reply by yuanweize** [reply]
> @yuanweize 
> 
> 
> 
> 另外，可能需要你帮个忙：我在考虑切换某些 CDN 环境，方便的话，帮忙测试下。
> 
> 
> 
> 请问你访问 https://cdn.staticfile.org/font-awesome/6.2.0/css/fontawesome.min.css 是否正常？同时请告知速度如何？

欧洲 秒开

**Reply by Kenshin** [reply]
@yuanweize 

收到~  🙏
